File: easy/average-salary-excluding-the-minimum-and-maximum-salary/average-salary-excluding-the-minimum-and-maximum-salary.js
/**
 * @param {number[]} salary
 * @return {number}
 */
var average = function (salary) {
  salary.sort((a, b) => a - b);
  salary.splice(0, 1);
  salary.splice(-1, 1);
  const sum = salary.reduce((p, c) => p + c);
  return sum / salary.length;
};

File: easy/average-salary-excluding-the-minimum-and-maximum-salary/average-salary-excluding-the-minimum-and-maximum-salary.c
double average(int *salary, int salarySize)
{
    int min, max, sum;
    min = max = sum = salary[0];
    for (int i = 1; i < salarySize; i++)
    {
        sum += salary[i];
        if (salary[i] < min)
        {
            min = salary[i];
        }
        if (salary[i] > max)
        {
            max = salary[i];
        }
    }
    printf("%i", sum);
    return 1.0 * (sum - max - min) / (salarySize - 2);
}